xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateStore.java

```diff
@@ -1368,17 +1369,20 @@ private void loadXWikiCollectionInternal(BaseCollection object1, XWikiDocument d
                 try {
                     if ((bclass != null) && (bclass.hasCustomMapping()) && context.getWiki().hasCustomMappings()) {
                         Session dynamicSession = session.getSession(EntityMode.MAP);
-                        Object map = dynamicSession.load(bclass.getName(), object.getId());
+                        String className = this.localEntityReferenceSerializer.serialize(bclass.getDocumentReference());
+                        @SuppressWarnings("unchecked")
+                        Map<String, ?> map = (Map<String, ?>) dynamicSession.load(className, object.getId());
                         // Let's make sure to look for null fields in the dynamic mapping
-                        bclass.fromValueMap((Map) map, object);
-                        handledProps = bclass.getCustomMappingPropertyList(context);
-                        for (String prop : handledProps) {
-                            if (((Map) map).get(prop) == null) {
-                                handledProps.remove(prop);
+                        bclass.fromValueMap(map, object);
+                        for (String prop : bclass.getCustomMappingPropertyList(context)) {
+                            if (map.get(prop) != null) {
+                                handledProps.add(prop);
                             }
                         }
                     }
-                } catch (Exception e) {
+                } catch (HibernateException e) {
+                    this.logger.error("Failed loading custom mapping for doc [{}], class [{}], nb [{}]", 
+                            object.getDocumentReference(), object.getXClassReference(), object.getNumber(), e);
```

Comment:
Question: shouldn't we throw an exception if custom mapping failed to load? Isn't that an important enough problem?

PS: I know this is not strictly related to your PR since what was there before was even worse, but it got me wondering.

Reply:
Perhaps a `HibernateException` is thrown in case a class mapping is set to internal, but no actual hibernate mapping is given for the class? I'm not sure if that is the case.
If yes, a valid fallback would be to continue and load from `xwikiproperties`.
If no, I would expect the returned map to be empty and the exception should be wrapped and re-thrown in the catch block.